### PR TITLE
Stream and advertise metric correlations to the cloud

### PR DESCRIPTION
**Pull request by MrZammler**
##### Summary

This PR turns MC in the agent to default off, streams the MC version to parents, and will send the status of MC through the capabilities field in UpdateNodeInfo messages.

##### Test Plan

We will test this PR with the cloud backend to make sure things are in place first.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

**Comment by MrZammler** [comment]
cc @juacker

File: aclk/aclk_util.h [hunk 1]
```diff
@@ -20,6 +20,7 @@
 
 // Helper stuff which should not have any further inside ACLK dependency
 // and are supposed not to be needed outside of ACLK
+#define ACLK_LOG_CONVERSATION_DIR "/tmp/aclk-messages"
```
Comment (stelfrag) [inline aclk/aclk_util.h:23]: Shouldn't this be removed?

File: database/metric_correlations.c [hunk 2]
```diff
@@ -4,7 +4,8 @@
 #include "KolmogorovSmirnovDist.h"
 
 #define MAX_POINTS 10000
-int enable_metric_correlations = CONFIG_BOOLEAN_YES;
+int enable_metric_correlations = CONFIG_BOOLEAN_NO;
```
Comment (stelfrag) [inline database/metric_correlations.c:7]: How is it enabled?

File: aclk/aclk_util.h (hunk 1 above)
Comment (MrZammler) [reply aclk/aclk_util.h]: Dumb, yes, sorry, removed...

File: database/metric_correlations.c (hunk 2 above)
Comment (MrZammler) [reply database/metric_correlations.c:7]: It is part of the previous PR, there is code to read it from the configuration -> https://github.com/netdata/netdata/blob/108bfb7918a86255bf79119f9b8eee604943ae76/daemon/main.c#L558

**Comment by juacker** [comment]
After some tests with @MrZammler, we've seen mc capability information is reaching the cloud and being applied perfectly.

**Review by underhood — APPROVED** [review]

**Review by thiagoftsm — APPROVED** [review]
I tested locally and I observed that cloud and streaming worked as expected.

**Comment by MrZammler** [comment]
Thanks a lot!

**Pull request merged**